pyfmodex/studio/library.py

```diff
@@ -15,10 +16,19 @@
 
 LIBRARY = None
 
-
 def get_library():
     """Load libary file according to architecture's ctype method."""
     global LIBRARY
     if not LIBRARY:
-        LIBRARY = library_type.LoadLibrary(LIBRARY_NAME)
+        try:
+            LIBRARY = library_type.LoadLibrary(LIBRARY_NAME)
+        except:
+            current_directory = os.path.dirname(os.path.realpath(sys.argv[0]))
+            try:
+                LIBRARY = library_type.LoadLibrary(os.path.join(current_directory, LIBRARY_NAME))
+            except:
+                raise RuntimeError(
+                    "Pyfmodex could not find the fmodstudio.dll in the following folders: " + str(
+                        sys.executable) + ", " + str(sys.path) + ", " + str(current_directory))
```

Comment:
You unfortunately can not get the operating system's library search path using sys.path, that's a python specific thing. On Windows, the system searches in all the directories listed in the PATH environment variable and Linux is more complex still, see https://unix.stackexchange.com/questions/22926/where-do-executables-look-for-shared-objects-at-runtime for example. And what Mac os does, honestly i have no idea.